### ISSUE TITLE
Support for 8+ Xbox controllers w/ DirectInput

### ISSUE THREAD

**Issue by iampremo** [YoYoGames/GameMaker-Bugs#3790]
User request message from ticket [Ticket #190241]

 

Feature Type: feat_addition_ingame
Description: Hello,

I got interested into games supporting more than 4 players on the same PC. It looks like it's not possible to use more than 4 Xbox controllers with the way GameMaker manages it.
Yet there are many games made with other engine that support more than 4 Xbox controllers (eg: Duck Game) because they use Direct Input instead of Xinput.

I've done some tests and it looks like it's impossible because there's no way to manually ask GameMaker to detect Xbox controllers through DirectInput instead of Xinput.

The only way is to use additional controllers which do not support Xinput (like PS4 controller) but that doesn't make things easy for the players...

I was talking about this with the developer of the popular "Hidden in Plain Sight" who is currently being recreated from scratch by the author with GameMaker (see screenshot)
Basically the Xbox controllers will fill slots 0 to 3 and any additional Xbox controller will not be detected at all.

Ideally we should:
- Have the option to force all Xbox controllers to use DirectInput
OR better:
- Detect the first four Xbox controllers with Xinput and any additional ones with DirectInput (no idea if this is technically possible)

Is there any workaround? If not, is there a ticket about this?
Benefit: I'm also interested to make a 8 player game myself in the future and I'm hoping this issue could be fixed.
